src/main/java/com/conveyal/gtfs/validator/ServiceValidator.java

```diff
@@ -181,6 +181,8 @@ public void complete(ValidationResult validationResult) {
             LocalDate firstDate = LocalDate.MAX;
             LocalDate lastDate = LocalDate.MIN;
             for (LocalDate date : dateInfoForDate.keySet()) {
+                // If the date is invalid, skip.
+                if (date == null) continue;
```

Comment:
How is this error happening?  Shouldn't the computeIfAbsent method for adding keys and values never yield a null key?  It'd be very helpful to add a failing test case to illustrate what is causing the problem here.

Reply:
@evansiroky, there are now tests that load feeds with bad date values. When you run `canLoadFeedWithBadDates`, you'll see that it does encounter a null key.